src/loader.rs

```diff
@@ -25,28 +53,196 @@ impl AssetLoader for ObjLoader {
     }
 }
 
+#[cfg(feature = "scene")]
+impl FromWorld for ObjLoader {
+    fn from_world(world: &mut World) -> Self {
+        let supported_compressed_formats = match world.get_resource::<RenderDevice>() {
+            Some(render_device) => CompressedImageFormats::from_features(render_device.features()),
+
+            None => CompressedImageFormats::all(),
+        };
+        Self {
+            supported_compressed_formats,
+        }
+    }
+}
+
 #[derive(Error, Debug)]
 pub enum ObjError {
     #[error("Invalid OBJ file: {0}")]
-    InvalidFile(#[from] tobj::LoadError),
+    TobjError(#[from] tobj::LoadError),
+    #[cfg(feature = "scene")]
+    #[error("Invalid image file for texture: {0}")]
+    InvalidImageFile(PathBuf),
+    #[error("Asset reading failed: {0}")]
+    AssetIOError(#[from] bevy_asset::AssetIoError),
+    #[error("Texture conversion failed: {0}")]
+    TextureError(#[from] bevy_render::texture::TextureError),
 }
 
 async fn load_obj<'a, 'b>(
     bytes: &'a [u8],
     load_context: &'a mut LoadContext<'b>,
+    #[cfg(feature = "scene")] supported_compressed_formats: CompressedImageFormats,
 ) -> Result<(), ObjError> {
-    let mesh = load_obj_from_bytes(bytes)?;
-    load_context.set_default_asset(LoadedAsset::new(mesh));
+    #[cfg(not(feature = "scene"))]
+    let obj = load_obj_from_bytes(bytes)?;
+    #[cfg(feature = "scene")]
+    let obj = load_obj_from_bytes(bytes, load_context, supported_compressed_formats).await?;
+
+    load_context.set_default_asset(LoadedAsset::new(obj));
     Ok(())
 }
 
-fn load_mtl(_path: &std::path::Path) -> tobj::MTLLoadResult {
-    Err(tobj::LoadError::OpenFileFailed)
+#[cfg(feature = "scene")]
+async fn load_texture_image<'a, 'b>(
+    image_path: &'a str,
+    load_context: &'a mut LoadContext<'b>,
+    supported_compressed_formats: CompressedImageFormats,
+) -> Result<(Image, String), ObjError> {
+    let path = Path::new(&image_path);
+    let filename = path
+        .to_str()
+        .ok_or(ObjError::InvalidImageFile(path.to_path_buf()))?;
+    let extension = ImageType::Extension(
+        path.extension()
+            .and_then(|e| e.to_str())
+            .ok_or(ObjError::InvalidImageFile(path.to_path_buf()))?,
+    );
+    let bytes = load_context.asset_io().load_path(path).await?;
+    // TODO(luca) confirm value of is_srgb
+    let is_srgb = true;
+    Ok((
+        Image::from_buffer(&bytes, extension, supported_compressed_formats, is_srgb)?,
+        filename.to_string(),
+    ))
+}
+
+#[cfg(feature = "scene")]
+async fn load_obj_from_bytes<'a, 'b>(
+    mut bytes: &'a [u8],
+    load_context: &'a mut LoadContext<'b>,
+    supported_compressed_formats: CompressedImageFormats,
+) -> Result<Scene, ObjError> {
+    let options = tobj::GPU_LOAD_OPTIONS;
+    let asset_io = &load_context.asset_io();
+    let obj = tobj::load_obj_buf_async(&mut bytes, &options, |p| async move {
+        asset_io
+            .load_path(Path::new(&p))
```

Comment:
And about the path: this should also use a relative path to the original asset path. For example, this whole code block could be written as:

```
    log::info!("Loading Obj from  path = {:?}  size = {} bytes", asset_path, bytes.len());
    let obj = tobj::load_obj_buf_async(&mut bytes, &options, |p| async move {
        {
            let material_path: &Path = asset_path.as_ref();
            let mut material_path = material_path.to_owned();
            material_path.set_file_name(p);
            log::info!("Loading Obj Material: {:?}", &material_path);

            asset_io
            .load_path(&material_path)
            .await
            .map_or(Err(tobj::LoadError::OpenFileFailed), |bytes| {
                tobj::load_mtl_buf(&mut bytes.as_slice())
            })
        }
    })
    .await?;
```